packages/generator/tests/test_assets/example_renders/measurement_plugin_client/void_measurement_client.py

```diff
@@ -0,0 +1,221 @@
+
+    @property
+    def sites(self) -> List[int]:
+        """The sites where the measurement must be executed."""
+        if self._pin_map_context is not None:
+            return self._pin_map_context.sites
+
+    @sites.setter
+    def sites(self, val: List[int]) -> None:
+        if self._pin_map_context is None:
+            raise AttributeError(
+                "Cannot set sites because the pin map context is None. Please provide a pin map context or register a pin map before setting sites."
+            )
+        self._pin_map_context = self._pin_map_context._replace(sites=val)
+
+    def _get_stub(self) -> v2_measurement_service_pb2_grpc.MeasurementServiceStub:
+        if self._stub is None:
+            with self._initialization_lock:
+                if self._stub is None:
+                    service_location = self._get_discovery_client().resolve_service(
+                        provided_interface=_V2_MEASUREMENT_SERVICE_INTERFACE,
+                        service_class=self._service_class,
+                    )
+                    channel = self._get_grpc_channel_pool().get_channel(
+                        service_location.insecure_address
+                    )
+                    self._stub = v2_measurement_service_pb2_grpc.MeasurementServiceStub(channel)
+        return self._stub
+
+    def _get_discovery_client(self) -> DiscoveryClient:
+        if self._discovery_client is None:
+            with self._initialization_lock:
+                if self._discovery_client is None:
+                    self._discovery_client = DiscoveryClient(
+                        grpc_channel_pool=self._get_grpc_channel_pool(),
+                    )
+        return self._discovery_client
+
+    def _get_grpc_channel_pool(self) -> GrpcChannelPool:
+        if self._grpc_channel_pool is None:
+            with self._initialization_lock:
+                if self._grpc_channel_pool is None:
+                    self._grpc_channel_pool = GrpcChannelPool()
+        return self._grpc_channel_pool
+
+    def _get_pin_map_client(self) -> PinMapClient:
+        if self._pin_map_client is None:
+            with self._initialization_lock:
+                if self._pin_map_client is None:
+                    self._pin_map_client = PinMapClient(
+                        discovery_client=self._get_discovery_client(),
+                        grpc_channel_pool=self._get_grpc_channel_pool(),
+                    )
+        return self._pin_map_client
+
+    def _create_file_descriptor(self) -> None:
+        create_file_descriptor(
+            input_metadata=list(self._configuration_metadata.values()),
+            output_metadata=list(self._output_metadata.values()),
+            service_name=self._service_class,
+            pool=descriptor_pool.Default(),
+        )
+
+    def _create_measure_request(
+        self, parameter_values: List[Any]
+    ) -> v2_measurement_service_pb2.MeasureRequest:
+        serialized_configuration = any_pb2.Any(
```

Comment:
we would need to add type url here once the other PR for [C# service support](https://github.com/ni/measurement-plugin-python/pull/889/) goes in.